go/vt/vttablet/grpctmclient/client.go

```diff
@@ -648,17 +662,17 @@ func (client *Client) InitReplica(ctx context.Context, tablet *topodatapb.Tablet
 }
 
 // DemoteMaster is part of the tmclient.TabletManagerClient interface.
-func (client *Client) DemoteMaster(ctx context.Context, tablet *topodatapb.Tablet) (string, error) {
+func (client *Client) DemoteMaster(ctx context.Context, tablet *topodatapb.Tablet) (string, *replicationdatapb.MasterStatus, error) {
 	cc, c, err := client.dial(tablet)
 	if err != nil {
-		return "", err
+		return "", nil, err
 	}
 	defer cc.Close()
 	response, err := c.DemoteMaster(ctx, &tabletmanagerdatapb.DemoteMasterRequest{})
 	if err != nil {
-		return "", err
+		return "", nil, err
 	}
-	return response.Position, nil
+	return response.DeprecatedPosition, response.MasterStatus, nil
```

Comment:
We currently only guarantee RPCs will work with client/server version skew if the client is older, but we plan to allow the client to be newer and we'd like to start getting experience with doing that.

In this case, I think we could facilitate compatibility by checking if `response.MasterStatus` is nil, and assume the response came from an older server if so. We could create a `MasterStatus` from scratch in that case and fill in `MasterStatus.Position` from `response.DeprecatedPosition`.

This is only half of ensuring compatibility in this direction though. The other half is we'd have to make sure any callers of this function know that `MasterStatus.FilePosition` might be nil, and they need to have some fallback logic to deal with that, if they use that field.

Reply:
Done.